R/arc-read.R

```diff
@@ -286,3 +296,88 @@ label_layer_fields <- function(
   attr(x, "label") <- value
   x
 }
+
+
+#' Set coded values for FeatureLayer or Table data frame
+#'
+#' [set_layer_coded_values()] can replace column values based on `codedValue`
+#' type field domains from a corresponding `Table` or `FeatureLayer` object
+#' created with `arc_open()`.
+#'
+#' @param .data A data frame returned by `arc_select()` or `arc_read()`.
+#' @param .layer A Table or FeatureLayer object. Required.
+#' @param field Default `NULL`. Field or fields to replace. Fields that are do
+#'   not have coded values domains are ignored.
+#' @param codes Use of field alias values. Default `c("replace"),`.
+#' There are two options:
+#'
+#'  - `"replace"`: coded values replace existing column values.
+#'  - `"label"`: coded values are applied as value labels.
+#' @inheritParams rlang::args_error_context
+#' @export
+set_layer_coded_values <- function(
+    .data,
+    .layer,
+    field = NULL,
+    codes = c("replace", "label"),
+    call = rlang::caller_env()) {
+  values <- pull_coded_values(.layer, field = field)
+
+  # Check if coded values is an empty list
+  if (rlang::is_empty(values)) {
+    if (is.null(field)) {
+      cli::cli_warn(
+        "{.arg layer} does not contain any coded values."
+      )
+    } else {
+      cli::cli_warn(
+        "{.arg field} does not specific any coded value fields."
```

Comment:
typo, but I'm wondering what the difference is between these two branches. Shouldn't there _always_ be a warning when `values` is empty?

Reply:
If a user specifies a field (or fields) that _isn't_ a coded value domain field, they should get an error specific to those fields (and maybe should specify a different field). If field is `NULL` and values is empty, they shouldn't bother with this function at all. I can clean-up the logic though - it is just there to make sure the warning is more specific.

Reply:
ah, that makes sense! thank you :)